### ISSUE TITLE
How to reassemble audio from stft()?

### ISSUE THREAD

**Issue by jbeuckm** [soumith/torch-signal#6]
Is there a function that accepts the spectrum frames from stft() and reassembles the audio signal?

**Comment by soumith** [comment]
At the moment i didnt write the inverse stft. But the source code is fairly
simple, see if you can give it a shot.

**Comment by jfsantos** [comment]
Here's a sample implementation I translated quickly from my Python code to test some things. (note: might be broken, I didn't test it exhaustively with different window/overlap combinations). Reconstruction won't be perfect unless the input and windows are properly chosen (i.e., you will need zero-padding and choosing the correct window for the amount of overlap you use).

```lua
function istft(X, win, hop)
    local x = torch.zeros((X:size(1)-1)*hop + win)
    framesamp = X:size(2)
    hopsamp = hop
    for n=1,X:size(1) do
        i = 1 + (n-1)*hopsamp
        print(i, i + framesamp - 1)
        x[{{i, i + framesamp - 1}}] = x[{{i, i + framesamp - 1}}] + complex.real(signal.ifft(X[n]))
    end
    return x
end
```
